Caltech-IPAC/firefly
# DM-13273: upgrade react to 16 and upgrade other packages

Upgrade react to 16 and upgrade other packages  

  - removed `chartEngine` app option, and references to `highcharts`
  - removed `react-component-resizable`, all code now using `react-sizeme`
  - removed `react-addons-perf`, it is not longer supports for 16
  - fixed a couple of files that had warning issues
  - new version of `validator` now only takes strings, fixed any code using it
  - removed ImageViewerView because it was no longer necessary with SizeMe
  - a little code clean up if I was already working in the file.


_To Test:_

 - Delete your node_modules directory.
 - Choose area of the code you know about.
 - Make sure it behaves like before.
 - Pay attention to any UI subtleties: resizing, updating correctly, layout, etc
